core/src/main/java/org/apache/carbondata/core/view/MVProvider.java

```diff
@@ -105,10 +89,14 @@ private SchemaProvider getSchemaProvider(String databaseName) {
         if (schemaProvider == null) {
           String databaseLocation;
           if (databaseNameUpper.equalsIgnoreCase(CarbonCommonConstants.DATABASE_DEFAULT_NAME)) {
-            databaseLocation = CarbonUtil.checkAndAppendHDFSUrl(this.storeLocation);
+            databaseLocation = CarbonUtil.checkAndAppendHDFSUrl(
+                viewManager.getDatabaseLocation(databaseName)
```

Comment:
Please move getting database location before ifElse statement and reuse it in if and else block

Reply:
OK